File: frontend/src/types/elderly.ts
export interface CallSchedule {
  enabled: boolean;
  times: string[];  // ["09:00", "14:00"]
}

export interface Elderly {
  id: number;
  caregiver_id: number;
  name: string;
  age?: number;
  phone?: string;
  call_schedule: CallSchedule;
  health_condition?: string;
  medications?: Array<{ name: string; dosage: string; frequency: string }>;
  emergency_contact?: string;
  risk_level: string;
  notes?: string;
  created_at: string;
  updated_at: string;
}

export interface ElderlyCreateRequest {
  name: string;
  age?: number;
  phone?: string;
  call_schedule?: CallSchedule;
  health_condition?: string;
  medications?: Array<{ name: string; dosage: string; frequency: string }>;
  emergency_contact?: string;
  notes?: string;
}

File: iOS/Somi/Models/Elderly.swift
import Foundation

struct CallSchedule: Codable {
    let enabled: Bool
    let times: [String]
}

struct Elderly: Codable, Identifiable {
    let id: Int
    let caregiver_id: Int
    let name: String
    let age: Int?
    let phone: String?
    let call_schedule: CallSchedule
    let health_condition: String?
    let medications: [Medication]?
    let emergency_contact: String?
    let risk_level: String
    let notes: String?
    let created_at: String
    let updated_at: String
}

struct Medication: Codable {
    let name: String
    let dosage: String
    let frequency: String
}

struct ElderlyCreateRequest: Codable {
    let name: String
    let age: Int?
    let phone: String?
    let call_schedule: CallSchedule?
    let health_condition: String?
    let medications: [Medication]?
    let emergency_contact: String?
    let notes: String?
}
